build/makelib/golang.mk

```diff
@@ -170,7 +170,13 @@ go.mod.update:
 
 .PHONY: go.mod.check
 go.mod.check:
-	@echo === ensuring modules are tidied
+	@echo === syncing root modules with APIs modules
+	@cp -a go.sum pkg/apis/go.sum
+	@cat go.mod | sed -e 's|^module github.com/rook/rook|module github.com/rook/rook/pkg/apis|' \
+	                  -e '\:^replace github.com/rook/rook/pkg/apis => ./pkg/apis:d' > pkg/apis/go.mod
+	@echo === ensuring APIs modules are tidied
+	@(cd pkg/apis/; $(GOHOST) mod tidy -compat=1.17)
```

Comment:
Rook is building with go 1.19, do we still need this compat flag?

Reply:
It depends if rook whats to support go 1.17 if not I would keep the golang but change it's value to 1.19

Reply:
Rook attempts to maintain compatibility with the two most recent golang versions, so 1.19 is the min version currently

Reply:
fixed